corehq/ex-submodules/casexml/apps/case/util.py

```diff
@@ -184,34 +184,37 @@ def iter_cases(case_ids, strip_history=False, wrap=True):
             yield case
 
 
-def get_datetime_case_property_changed(case, case_property_name, value):
+def get_case_property_changed_info(case, case_property_name, value):
     """Returns the datetime a particular case property was changed to a specific value
 
     Not performant!
     """
     from casexml.apps.case.xform import get_case_updates
-    PropertyChangedInfo = namedtuple("PropertyChangedInfo", 'new_value modified_on')
+    PropertyChangedInfo = namedtuple("PropertyChangedInfo", 'case_update new_value modified_on')
 
     def property_changed_in_action(action, case_property_name):
         update_actions = [
-            (update.modified_on_str, update.get_update_action())
+            (update.modified_on_str, update.get_update_action(), action)
             for update in get_case_updates(action.form)
             if update.id == case.case_id
         ]
-        for (modified_on, action) in update_actions:
-            if action:
-                property_changed = action.dynamic_properties.get(case_property_name)
+        for (modified_on, update_action, case_update) in update_actions:
```

Comment:
it's confusing to me that this is called `action` when this list is created, but `case_update` here. Which is it?

Reply:
oh yes.. this is very confusing. 
Improving.